certificate-manager/src/test/java/io/strimzi/certs/IpAndDnsValidationTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class IpAndDnsValidationTest {
+    @Test
+    public void testIPv6()   {
```

Comment:
Other cases:
* A group with > 4 chars
* Presence of a triple `:::`
* Too many/few groups in presence of `::`